exporter/datadogexporter/metrics_exporter.go

```diff
@@ -66,6 +70,36 @@ func newMetricsExporter(ctx context.Context, params component.ExporterCreateSett
 	return &metricsExporter{params, cfg, ctx, client, tr}
 }
 
+func (exp *metricsExporter) pushSketches(ctx context.Context, sl sketches.SketchSeriesList) error {
+	payload, err := sl.Marshal()
+	if err != nil {
+		return fmt.Errorf("failed to marshal sketches: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx,
+		http.MethodPost,
+		exp.cfg.Metrics.TCPAddr.Endpoint+sketches.SketchSeriesEndpoint,
+		bytes.NewBuffer(payload),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to build sketches HTTP request: %w", err)
+	}
+
+	utils.SetDDHeaders(req.Header, exp.params.BuildInfo, exp.cfg.API.Key)
+	utils.SetExtraHeaders(req.Header, utils.ProtobufHeaders)
+	resp, err := exp.client.HttpClient.Do(req)
+
+	if err != nil {
+		return fmt.Errorf("failed to do sketches HTTP request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
```

Comment:
This is fine if no retries are performed. If retried were performed we'd want to no retry for some specific 4xx status codes, see datadog-agent logic [here](https://github.com/DataDog/datadog-agent/blob/7abd230002d930dd42f14a65d424191075b1df61/pkg/forwarder/transaction/transaction.go#L337-L348)

Reply:
👍 this is also an issue for 'normal' metrics, we are tracking this internally on AP-1252